### DISCUSSION TITLE
question?.:

### DISCUSSION THREAD

**Discussion by StackTrace18** [nrbnlulu/strawberry-django-auth#221] [Q&A] [answered]
Hi All, where can I ask questions about this library? thanks.

**Comment by nrbnlulu** [comment, marked as answer]
Here :smile: !

**Reply by StackTrace18** [reply]
How do I include custom graphql errors in GQLAuthError, or does everyone do it some other way? 
Btw, loving this library, makes auth in django easy and powerful. Thanks.

**Reply by nrbnlulu** [reply]
What do you want to add up?
But generally, you would need to subclass the directive you are using i.e
https://github.com/nrbnlulu/strawberry-django-auth/blob/b7b62da4bb5ed25786fb50bf74255933944e7d38/gqlauth/core/directives.py#L40

and return whatever you want...
```py
class MyError(GQLAuthError):
    ...

class MyDirective(TokenRequired):
    def resolve_permission(
        self, source: Any, info: Info, args: list, kwargs: dict
    ) -> Optional[GQLAuthError]:
        res  = super().resolve_permission(source, info, args, kwargs)
        return MyError()

@strawberry.type
class Query:
    @field(directives=[MyDirective()])
    def auth_entry(self) -> Union[MyError, MyAuthorizedQueries]:
        return MyAuthorizedQueries()
```

**Reply by StackTrace18** [reply]
ok, i'll just handle errors on the front end. Thanks.
